lib/firestore/firestore_gateway.dart

```diff
@@ -133,9 +139,20 @@ class FirestoreGateway {
             : null);
   }
 
+  // return call options for using a service account.
+  CallOptions _serviceAccountChannelOptions() {
+    var gapp = Platform.environment['GOOGLE_APPLICATION_CREDENTIALS'];
+    var _saJson = File(gapp).readAsStringSync();
+    var _jwt = JwtServiceAccountAuthenticator(_saJson);
+    return _jwt.toCallOptions;
+  }
```

Comment:
I feel like this belongs in the firebase_auth code rather than in the Firestore.
Ideally your changes shouldn't need to touch the Firestore code at all.

Also, while reading the credentials path from environment variables is a good practice, I don't think the library should dictate it.
A way to avoid that is to have the library accept the credentials path as an initialisation argument - how the path is determined is left up to the user (maybe demonstrate reading from env variables in the example).